core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowVersionResource.scala

```diff
@@ -31,13 +31,62 @@ import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
 
 object WorkflowVersionResource {
   final private lazy val context = SqlServer.createDSLContext()
+  final private val workflowVersionDao = new WorkflowVersionDao(context.configuration)
   final private val workflowDao = new WorkflowDao(context.configuration)
   // constant to indicate versions should be aggregated if they are within the specified time limit
   private final val AGGREGATE_TIME_LIMIT_MILLSEC =
     AmberUtils.amberConfig.getInt("user-sys.version-time-limit-in-minutes") * 60000
   // list of Json keys in the diff patch that are considered UNimportant
   private final val VERSION_UNIMPORTANCE_RULES = List("/operatorPositions/")
 
+  /**
+    * This function retrieves the latest version of a workflow
+    * @param wid
+    * @return vid
+    */
+  def getLatestVersion(wid: UInteger): UInteger = {
+    context
+      .select(WORKFLOW_VERSION.VID)
+      .from(WORKFLOW_VERSION)
+      .leftJoin(WORKFLOW)
+      .on(WORKFLOW_VERSION.WID.eq(WORKFLOW.WID))
+      .where(WORKFLOW_VERSION.WID.eq(wid))
+      .fetchInto(classOf[UInteger])
+      .toList
+      .max
+  }
+
+  /**
+    * This function inserts a new version for the current workflow
+    * @param patch
+    * @param wid
+    */
+  def insertVersion(patch: String, wid: UInteger): Unit = {
```

Comment:
I understand the semantically the `patch` can be viewed as a version. However, this API is like a lower api on `insertDiff(patch)` and it is being called in other resources like `WorkflowResource`. 
Can we do something like this instead:

in `WorkflowResource`:
```
def persistWorkflow(uid, wid, content):

if usercheck(uid):
     oldWorkflow = workflowDao.fetch(wid)
     workflowDao.insert(workflow)
     workflowVersionResource.insertOrUpdate(oldWorkflow, workflow)
```
and in `WorkflowVersionResource`:
```
def insertOrUpdate(oldWorkflow, newWorkflow):
      patch = diff(oldWorkflow, newWorkflow)
      insertDiff(patch)
      insertDiff("[]")
```
      
so that the versioning logic can be self contained in the version resource, and the user of it won't need to worry about the internal `patch`?

Reply:
done